### ISSUE TITLE
TextArea missing readonly prop

### ISSUE THREAD

**Issue by jmroon** [carbon-design-system/carbon-components-svelte#1009]
While TextInput has a readonly prop, TextArea appears to be missing it.

edit: Well, correction, the prop exists (appears to be passed in through $$restProps), but it's undocumented and the component is not properly styled when readonly. Also, even if readonly is set to false, you can only type in 1 character and it then doesn't allow for any more input.

**Comment by metonym** [comment]
Toggling `readonly={false}` does indeed cause a bug; the workaround in https://github.com/carbon-design-system/carbon-components-svelte/pull/719 no longer works. This needs fixing and documentation.

Regarding the styles: that would be an issue for Carbon.

**Comment by jmroon** [comment]
I was able to make it consistent with:

```
// Form.svelte
<script>
  export let editing = false;
  $: readonly = editing ? undefined : true;
</script>
<TextArea light={!editing} {readonly} />
```

**Comment by metonym** [comment]
Fixed in [v0.56.0](https://github.com/carbon-design-system/carbon-components-svelte/releases/tag/v0.56.0)

Note: Carbon does not yet have a corresponding style for a `TextArea` readonly variant. Your solution of toggling the `light` prop is a pragmatic workaround.
